### ISSUE TITLE
[Feature Request] Order/Sort Page Collection by multiple criteria

### ISSUE THREAD

**Issue by NicoHood** [getgrav/grav#3073]
I found no easy way to sort a page by multiple criteria. For example I want to sort a collection based in `header.rating.average` first and second I want to sort them by name, or date or other options.

I did not even find an easy way for a workaround. Is there a chance to add this feature to grav?

**Comment by w00fz** [comment]
You should consider playing around with grav 1.7 and flex pages where a lot of work has been done to ease these kind of operations. 
Flex pages is also a type of flex objects so filtering and sorting is now more advanced.

https://learn.getgrav.org/17/advanced/flex/using/collection

Keep also in mind it won’t be long for 1.7 to be released in the stable channel (https://getgrav.org/blog/grav-17-stable-jan-2020).

**Comment by NicoHood** [comment]
That is a good note. The last time I checked flex object (~2 Month ago) there was a note, that using flex pages (in contrast to flex objects) is not recommended, as it is currently (at that time) too slow. Also there was almost no documentation. That that change in the meantime?

By the way: I completely missed that blog entry. It seems my thunderbird cannot load new rss feed entries. All other feeds work properly. I've readded the feed and got the new messages now. Not sure if it will work when new blog posts pop up. Weird...

**Comment by mahagr** [comment]
Instead of pages I would just create my custom directory for the restaurants and use it. As you can create your own methods to the collections and objects, everything should be possible.

**Comment by NicoHood** [comment]
I am still confused. How can I render a flex object as single page? Each restaurant should have its own page/url. I thought that flex pages are not stable. I think I am understanding things wrong.

**Comment by NicoHood** [comment]
I think a skeleton to demonstrate the flex objects would help a lot! I tried the example contact directory. But how can I make each contact render as separate page?

**Comment by mahagr** [comment]
* **Specific URL:** I just added support for passing id into the page header with some documentation (will push soon) in case if you want to have single object shown in the page. It allows you to have fully custom URLs for the objects, in case if you need them. It is also possible to define key field from the blueprint (DOCS NEEDED), which allows you to have: `id:pizzarestaurant`.
* **Page context:** I don't recommend sending the whole page object as a parameter as it prevents caching. Just send route instead and fetch the page inside the template.
* **ID length:** IDs are random 16 letters by default, it can be changed from the blueprints. ID can also be set to a slug if needed.
* **Template per type:** Yes, I will add docs on how to do this shortly (already added the examples).
* **Title:** Already added support for this.
* **Flex Pages:** It's used by default in admin, but not in the frontend. It is available in the frontend, though.

**Comment by mahagr** [comment]
See: https://learn.getgrav.org/17/advanced/flex/using#using-flex-objects-page-type

PS: you need to download develop version of the plugin from here: https://github.com/trilbymedia/grav-plugin-flex-objects/

**Comment by NicoHood** [comment]
Alright, so "Flex Pages" refers to the `flex-objects.md`? If that is the case, and this is called "unstable" what can the flex objects be used for, if not for displaying page content? Maybe it would make sense to draft a few scenarios/usecases in which it would make sense to add flex pages, and how a migration could look like. Maybe a few in theory and one in details (for example migrating the blog into flex objects instead of pages). Just an Idea...

Thanks for your time :-)

**Comment by mahagr** [comment]
No, it does not refer to `flex-objects.md` which is a page type. It refers to a new implementation of the whole pages subsystem in Grav. So everything that handles the routing and the pages.

Well, Flex Pages are being used exclusively in the admin. The old pages admin is still available (in case if something is not working), but the new admin is very stable already -- in fact in many cases it works better than the old one.
